### PR TITLE
Checkpoint rage control

### PR DESCRIPTION
**Pull request by lechuga16**
- Fixed a bug, when a survivor enters a team, the announcement is displayed.
- Rearranged the code to make it simpler to review.
- `Event_RoundEndEvent `was replaced by `L4D2_OnEndVersusModeRound_Post`.
- Updated the syntax

Note: I made these changes in order to identify the '`if (LoadFromAddress(pTarget, NumberType_Int8) != CALL_OPCODE)`' error. although I can't figure out how to fix it.

### CODE REVIEW

**Comment by nikita1824** [comment]
Hello, we faced `if (LoadFromAddress(pTarget, NumberType_Int8) != CALL_OPCODE)` scenario and this is probably left4dhooks unloads before checkpoint-rage-control. 
Try to disable requirement of left4dhooks.

File: addons/sourcemod/scripting/checkpoint-rage-control.sp [hunk 1]
```diff
@@ -88,165 +96,262 @@ public void OnMapStart()
...
+		return;
+
 	HookEvent("player_entered_start_area", Event_EnteredStartArea);
-	HookEvent("round_end", Event_RoundEndEvent);
-	HookEvent("round_start", Event_RoundStart);
 	HookEvent("player_death", Event_PlayerDeath);
-	DebugPrint("Prepared hooks");
+	HookEvent("player_team", Event_PlayerTeam, EventHookMode_Pre);
+
+	DebugPrint("{blue}Prepared hooks{default} from L4D_OnSpawnTank_Post [player_entered_start_area, player_death]");
```
Comment (SirPlease) [inline addons/sourcemod/scripting/checkpoint-rage-control.sp:118]: Oi, `player_team` is also being hooked! 😛

**Review by SirPlease — CHANGES_REQUESTED** [review]
Just clear up the few typos/mistakes and it should be good to go 👍

**Comment by lechuga16** [comment]
@SirPlease done.

**Pull request merged**